bson/encode.go

```diff
@@ -60,6 +60,15 @@ var (
 	typeTimeDuration   = reflect.TypeOf(time.Duration(0))
 )
 
+var (
+	// spec for []uint8 or []byte encoding
+	arrayOps = map[string]bool{
```

Comment:
One nitpick here ( sorry! )
IIRC `map[string]struct{}` could be use here, and that one has an optimised lookup code when you want to only check for the presence of an element. 

this would transform the map lookups to 
```
if _, ok := arrayOps[name]; ok {
[...]
}
```
Which is not as clean. I'll let you make a call here. I haven't benchmarked the code, so I can't tell how much faster it is, but it is the idiomatic go way of doing things FWIW.

Reply:
I think the way will cause one more *alloc*, which can be avoid from previous.